linode/instanceip/framework_resource.go

```diff
@@ -54,13 +54,59 @@ func (r *Resource) Create(
 	isPublic := plan.Public.ValueBool()
 
 	client := r.Meta.Client
-	ip, err := client.AddInstanceIPAddress(ctx, linodeID, isPublic)
-	if err != nil {
-		resp.Diagnostics.AddError(
-			fmt.Sprintf("Failed to created instance (%d) IP", linodeID),
-			err.Error(),
-		)
-		return
+	var ip *linodego.InstanceIP
+	var err error
+
+	if !plan.Address.IsNull() && !plan.Address.IsUnknown() {
+		// Assign a reserved IP
+		createOpts := linodego.InstanceReserveIPOptions{
+			Type:    "ipv4",
+			Public:  isPublic,
+			Address: plan.Address.ValueString(),
+		}
+		_, err = client.AssignInstanceReservedIP(ctx, linodeID, createOpts)
+		if err != nil {
+			resp.Diagnostics.AddError(
+				fmt.Sprintf("Failed to assign reserved IP to instance (%d)", linodeID),
+				err.Error(),
+			)
+			return
+		}
+
+		// Fetch the IP information after assigning
```

Comment:
The code only needs to assign the IP, I think. Is there a need to read back and verify since you should be able to trust the return code of the first request?

Reply:
Yeah this part of the code was just for confirmation. I will be removing it in the next commit.